addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightDiscover.java

```diff
@@ -176,44 +183,102 @@ public void sendDiscover(ScheduledExecutorService scheduler) {
+                // We expect packets with a format like this: 10.1.1.27,ACCF23F57AD4,HF-LPB100
                 String[] msg = new String(buffer, 0, packet.getLength()).split(",");
-                if (msg.length >= 2 && msg[1].length() == 12) {
-                    // Stop resend timer if we got a packet.
-                    if (resendTimer != null) {
-                        resendTimer.cancel(true);
-                        resendTimer = null;
+
+                if (msg.length != 2 && msg.length != 3) {
+                    // That data packet does not belong to a Milight bridge. Just ignore it.
+                    continue;
+                }
+
+                int version = 3; // Assume version 3
+                // First argument is the IP
+                try {
+                    InetAddress.getByName(msg[0]);
+                } catch (UnknownHostException ignored) {
+                    // That data packet does not belong to a Milight bridge, we expect an IP address as first argument.
+                    // Just ignore it.
+                    continue;
+                }
+
+                // Second argument is the MAC address
+                if (msg[1].length() != 12) {
+                    // That data packet does not belong to a Milight bridge, we expect a MAC address as second argument.
+                    // Just ignore it.
+                    continue;
+                }
+
+                InetAddress addressOfBridge = ((InetSocketAddress) packet.getSocketAddress()).getAddress();
+                if (msg.length == 3) {
+                    version = 6; // It is probably version 6
+                    if ((msg[2].length() != 0 && !msg[2].equals("HF-LPB100"))) {
+                        logger.trace("Unexpected data. We expected a HF-LPB100 identifier {}", msg[2]);
+                        continue;
+                    }
+                    if (!checkForV6Bridge(addressOfBridge, msg[1])) {
+                        logger.trace("The device at IP {} does not seem to be a V6 Milight bridge", msg[0]);
+                        continue;
                     }
-                    // Determine version: Version 6 sends a third argument != ""
-                    int version = msg.length >= 3 && msg[2].trim().length() > 0 ? 6 : 3;
-                    // Notify all observers
-                    discoverResult.bridgeDetected(((InetSocketAddress) packet.getSocketAddress()).getAddress(), msg[1],
-                            version);
-                } else {
-                    logger.error("Unexpected data received {}", msg[0]);
                 }
+
+                stopResend();
+                discoverResult.bridgeDetected(addressOfBridge, msg[1], version);
             }
         } catch (IOException e) {
             if (willbeclosed) {
                 return;
             }
-            logger.error("{}", e.getLocalizedMessage());
+            logger.warn("{}", e.getLocalizedMessage());
```

Comment:
I don't know what message one might expect but some context could be useful, most likely / maybe the exceptions getLocalizedMessage does not even include the fact that the message belong to an IOException.